tsdb/index/tsi1/log_file.go

```diff
@@ -1074,8 +1074,8 @@ func (f *LogFile) seriesSketches() (sketch, tSketch estimator.Sketch, err error)
 }
 
 func (f *LogFile) Writes(entries []LogEntry) error {
-	f.mu.RLock()
-	defer f.mu.RUnlock()
+	f.mu.Lock()
+	defer f.mu.Unlock()
```

Comment:
Can you narrow the scope of the lock to just the loop?

Reply:
absolutely.